tests/actor-test.lisp

```diff
@@ -466,3 +479,31 @@
            (is-true init-called)
            (is (equalp '(1 2 3) (act-cell:state actor))))
       (ac:shutdown sys))))
+
+
+(test actor-of--limit-queue-size
+  "Tests that actor's queue size can be limited."
+  (let ((sys (asys:make-actor-system)))
+    (unwind-protect
+         (let* ((counter 0)
+                (queue-full-counter 0)
+                (actor (actor-of sys
+                                 :receive (lambda (msg)
+                                            (declare (ignore msg))
+                                            (sleep 1)
+                                            (incf counter))
+                                 :queue-size 1)))
+           (loop repeat 10
+                 do (handler-case (tell actor "run")
+                      (sento.queue:queue-full-error ()
+                        (incf queue-full-counter))))
+           
+           (sleep 2)
```

Comment:
Better use `await-cond` or some such here because we may have the result earlier and can spare some test runtime.
Possibly the 1 sec wait in `receive` can also be reduced.

Reply:
I've refactored code like [this](https://github.com/mdbergmann/cl-gserver/pull/91/commits/cdfdedd7f9e65c5ffacc452440796be1b83ca2f5#diff-eda80df249a13772d266d495d015b964f1e75568ee7f738d83accd15e6218984R497-R503), but it required some [changes](https://github.com/mdbergmann/cl-gserver/pull/91/commits/cdfdedd7f9e65c5ffacc452440796be1b83ca2f5#diff-edb4b3cb3533b5fae6cd4924a81f79e908ebf14b9032a4cc22d75502aaf7852fR101) in complete-p function.

Could you please review if it is ok to change `complete-p` this way? If so, probably a separate test should added for the cases where future can become completed with error.

What is interesting, I've replaced TELL with ASK and ASK does not raise `sento.queue:queue-full-error` error as TELL does. Instead it returns a "errored" future. Is this desired behaviour?

Reply:
`ask` always returns future. Insofar it consistently also delivers any errors as future result. While it's consistent it has a technical reason: `ask` spawn an anonymous actor that does the `tell` (and also waits for response via `reply`) so that the call to `ask` can be async.